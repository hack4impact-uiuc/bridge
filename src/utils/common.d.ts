declare module "@hack4impact-uiuc/bridge/common" {
  import { Props, ReactType } from "react";
  import {
    ColorProps,
    SpaceProps,
    LayoutProps,
    TypographyProps as StyledTypographyProps,
    BordersProps as StyledBordersProps,
    BoxShadowProps,
  } from "styled-system";

  export interface BaseProps extends Props<any> {
    as?: ReactType;
    className?: string;
  }
  export interface CommonProps
    extends BaseProps,
      ColorProps,
      SpaceProps,
      LayoutProps {}

  export interface TypographyProps extends BaseProps, StyledTypographyProps {}

  export interface BorderProps extends StyledBordersProps, BoxShadowProps {}
}
