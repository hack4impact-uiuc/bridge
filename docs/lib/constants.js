export const COMPONENTS_URL = '/components';
export const DESIGN_URL = '/design';
