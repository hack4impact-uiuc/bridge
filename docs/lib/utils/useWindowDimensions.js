import { useState, useEffect } from 'react';

function getWindowDimensions() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useWindowDimensions() {
  const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());

  useEffect(() => {
    function handleResize() {
      const { innerWidth: width, innerHeight: height } = window;
      setWindowDimensions({ width, height });
    }

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return windowDimensions;
}
