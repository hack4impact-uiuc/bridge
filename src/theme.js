/* eslint-disable prefer-destructuring */

function fontStack(fonts) {
  return fonts.map((font) => (font.includes(' ') ? `"${font}"` : font)).join(', ');
}

const DEFAULT_SHADOW = '0 1px 6px 2px rgba(0,0,0,0.15)';

const baseColors = {
  hackBlue: ['#B4D8FA', '#3E87CD', '#155DA1', '#0E4E8A', '#0D3F6E', '#031425'],
  indigoBlue: ['#BCD0F3', '#415F94', '#2D4979', '#253E68', '#13294E', '#061734'],
  impactGrey: ['#FAFBFC', '#EBEEF2', '#8B9199', '#666B72', '#46494F'],
};

const hackBluePalette = {
  lighter: baseColors.hackBlue[0],
  light: baseColors.hackBlue[1],
  primary: baseColors.hackBlue[2],
  dark: baseColors.hackBlue[3],
  darker: baseColors.hackBlue[4],
  text: baseColors.hackBlue[5],
};

const indigoPalette = {
  lighter: baseColors.indigoBlue[0],
  light: baseColors.indigoBlue[1],
  primary: baseColors.indigoBlue[2],
  dark: baseColors.indigoBlue[3],
  darker: baseColors.indigoBlue[4],
  text: baseColors.indigoBlue[5],
};

const impactGreyPalette = {
  lighter: baseColors.impactGrey[0],
  light: baseColors.impactGrey[1],
  midtone: baseColors.impactGrey[2],
  primary: baseColors.impactGrey[3],
  darker: baseColors.impactGrey[4],
};

const secondaryColors = {
  green: ['#BDF3C6', '#90E19E', '#4C9859', '#256830', '#154C1E', '#031D07'],
  red: ['#FFBCBF', '#D1595F', '#A62B31', '#8A1A1F', '#6B1014', '#35080A'],
  yellow: ['#FFECBD', '#FFD15F', '#FDBE21', '#DA9C03', '#A77700', '#201803'],
};

const greenPalette = {
  lighter: secondaryColors.green[0],
  light: secondaryColors.green[1],
  primary: secondaryColors.green[2],
  dark: secondaryColors.green[3],
  darker: secondaryColors.green[4],
  text: secondaryColors.green[5],
};

const redPalette = {
  lighter: secondaryColors.red[0],
  light: secondaryColors.red[1],
  primary: secondaryColors.red[2],
  dark: secondaryColors.red[3],
  darker: secondaryColors.red[4],
  text: secondaryColors.red[5],
};

const yellowPalette = {
  lighter: secondaryColors.yellow[0],
  light: secondaryColors.yellow[1],
  primary: secondaryColors.yellow[2],
  dark: secondaryColors.yellow[3],
  darker: secondaryColors.yellow[4],
  text: secondaryColors.yellow[5],
};

const typographyColors = {
  charcoal: '#272A2D',
  iron: '#4B4E52',
  ink: '#0A162A',
  slate: '#363949',
};

const colors = {
  black: '#000000',
  white: '#FFFFFF',
  ...baseColors,
  ...secondaryColors,
  ...typographyColors,
  hackBluePalette,
  indigoPalette,
  impactGreyPalette,
  greenPalette,
  redPalette,
  yellowPalette,
  text: {
    grayScale: {
      main: typographyColors.charcoal,
      support: typographyColors.iron,
    },
    coolTone: {
      main: typographyColors.ink,
      support: typographyColors.slate,
    },
  },
  // states or in other words color variants
  variants: {
    primary: hackBluePalette,
    success: greenPalette,
    error: redPalette,
    warning: yellowPalette,
    secondary: hackBluePalette,
    white: {
      primary: '#FFFFFF',
      dark: '#FFFFFF',
      lighter: indigoPalette.light,
    },
  },
};

const fontSizes = ['12px', '14px', '16px', '18px', '20px', '22px', '25px', '28px', '32px', '45px', '60px'];

const fonts = {
  chivo: fontStack(['Chivo']),
  hkGrotesk: fontStack(['HKGrotesk']),
};

const fontWeights = {
  light: 300,
  normal: 400,
  medium: 500,
  bold: 700,
};

// styling specific to the Button component
const button = {
  font: fonts.chivo,
  fontWeight: fontWeights.bold,
  fontSize: '14px',
  letterSpacing: '2px',
  // default values, based on the color palette
  default: (colorPalette) => ({
    bg: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
      disabled: colorPalette.lighter,
    },
    border: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colorPalette.light,
    },
    shadow: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    outline: {
      bg: {
        default: colors.white,
        hover: colors.white,
        disabled: colors.white,
      },
      border: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      fontColor: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      shadow: {
        default: 'none',
        hover: 'none',
        disabled: 'none',
      },
    },
  }),

  // everything below overrides the default values
  primary: {
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colors.white,
    },
  },
  white: {
    fontColor: {
      default: indigoPalette.primary,
      hover: indigoPalette.primary,
      disabled: indigoPalette.lighter,
    },
    shadow: {
      hover: DEFAULT_SHADOW,
    },
    outline: {
      bg: {
        default: indigoPalette.primary,
        hover: indigoPalette.primary,
        disabled: indigoPalette.primary,
      },
      border: {
        disabled: indigoPalette.light,
      },
      fontColor: {
        default: colors.white,
        hover: colors.white,
        disabled: indigoPalette.lighter,
      },
      shadow: {
        hover: DEFAULT_SHADOW,
      },
    },
  },
  secondary: {
    bg: {
      hover: hackBluePalette.darker,
      disabled: impactGreyPalette.light,
    },
    fontColor: {
      disabled: impactGreyPalette.midtone,
    },
    outline: {
      bg: {
        disabled: colors.white,
      },
      border: {
        hover: hackBluePalette.darker,
        disabled: impactGreyPalette.light,
      },
      fontColor: {
        disabled: indigoPalette.lighter,
      },
    },
  },
};


const link = {
  default: (colorPalette) => ({
    color: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
    },
  }),
  secondary: {
    default: hackBluePalette.primary,
    hover: hackBluePalette.darker,
  },
};


const theme = {
  // General
  colors,
  fonts,
  fontWeights,
  fontSizes,
  button,
  link,
  DEFAULT_SHADOW,
};


export default theme;
export { colors };
