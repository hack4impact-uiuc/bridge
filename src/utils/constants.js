import * as styledSystem from 'styled-system';
import systemPropTypes from '@styled-system/prop-types';
import PropTypes from 'prop-types';


const { compose, system } = styledSystem;

// Text Decoration Prop
export const textDecoration = system({
  prop: 'textDecoration',
  cssProperty: 'text-decoration',
});
textDecoration.propTypes = PropTypes.string;


// common style props to be passed into components
export const COMMON = compose(styledSystem.space, styledSystem.layout);
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.layout,
};

// for easier access
export const TYPOGRAPHY = styledSystem.typography;
export const LAYOUT = styledSystem.layout;
export const POSITION = styledSystem.position;
export const FLEX = styledSystem.flexbox;
export const GRID = styledSystem.grid;
export const COLOR = styledSystem.color;
export const BORDER = styledSystem.border;

TYPOGRAPHY.propTypes = systemPropTypes.typography;
LAYOUT.propTypes = systemPropTypes.layout;
POSITION.propTypes = systemPropTypes.position;
FLEX.propTypes = systemPropTypes.flexbox;
GRID.propTypes = systemPropTypes.grid;
COLOR.propTypes = systemPropTypes.color;
BORDER.propTypes = systemPropTypes.border;
