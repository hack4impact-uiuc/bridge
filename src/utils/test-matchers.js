/* eslint-disable react/forbid-foreign-prop-types */
// taken from https://github.com/primer/components/blob/master/src/utils/test-matchers.js
import 'jest-styled-components';

const stringify = (d) => JSON.stringify(d, null, '  ');
/**
 * These are props that styled-system aliases for backwards compatibility.
 * For some reason, they don't show up in our toImplementSystemProps() matcher,
 * so we skip over them.
 */
const ALIAS_PROP_TYPES = ['w', 'align', 'justify', 'wrap'];

// eslint-disable-next-line no-undef
expect.extend({
  toImplementSystemProps(Component, propNames) {
    const propKeys = new Set(Object.keys(Component.propTypes));
    const expectedPropKeys = Object.keys(propNames.propTypes);
    const missing = expectedPropKeys.filter((key) => !propKeys.has(key))
      .filter((key) => !ALIAS_PROP_TYPES.includes(key));
    return {
      pass: missing.length === 0,
      message: () => `Missing prop${missing.length === 1 ? '' : 's'}: ${stringify(missing)}`,
    };
  },
});
