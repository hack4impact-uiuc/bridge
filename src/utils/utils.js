import themeGet from '@styled-system/theme-get';
import * as styledSystem from 'styled-system';
import theme from '../theme';

const { get: getKey } = styledSystem;

export const get = (key) => themeGet(key, getKey(theme, key));

export const getOrDefault = (val, def) => (val !== undefined && val !== null ? val : def);

export const lodashGet = (object, path, defaultVal) => {
  const PATH = Array.isArray(path)
    ? path
    : path.split('.').filter((i) => i.length);
  if (!PATH.length) {
    return object === undefined ? defaultVal : object;
  }
  if (object === null || object === undefined || typeof (object[PATH[0]]) === 'undefined') {
    return defaultVal;
  }
  return lodashGet(object[PATH.shift()], PATH, defaultVal);
};
