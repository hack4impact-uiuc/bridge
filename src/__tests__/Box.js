// taken from https://github.com/primer/components/blob/master/src/__tests__/Box.js
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import { Box } from '..';
import { COMMON, LAYOUT } from '../utils/constants';
import theme from '../theme';
import { renderJSON } from '../utils/testing';
import 'babel-polyfill'; // axe violations required babel-polyfill

expect.extend(toHaveNoViolations);

describe('Box', () => {
  it('should have no axe violations', async () => {
    const { container } = render(<Box />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    cleanup();
  });

  it('implements system props', () => {
    expect(Box).toImplementSystemProps(LAYOUT);
    expect(Box).toImplementSystemProps(COMMON);
  });

  it('respects the "as" prop', () => {
    expect(renderJSON(<Box as="span" />).type).toEqual('span');
  });

  it('renders without any props', () => {
    expect(renderJSON(<Box />)).toMatchSnapshot();
  });

  it('renders margin', () => {
    expect(renderJSON(<Box m={1} theme={theme} />)).toMatchSnapshot();
    expect(renderJSON(<Box m={[0, 1, 2, 3]} theme={theme} />)).toMatchSnapshot();
    expect(renderJSON(<Box m={[1, 1, 1, 3]} theme={theme} />)).toMatchSnapshot();
  });

  it('renders padding', () => {
    expect(renderJSON(<Box p={1} theme={theme} />)).toMatchSnapshot();
    expect(renderJSON(<Box p={[0, 1, 2, 3]} theme={theme} />)).toMatchSnapshot();
    expect(renderJSON(<Box p={[1, 1, 1, 3]} theme={theme} />)).toMatchSnapshot();
  });

  it('respects display', () => {
    expect(renderJSON(<Box display="inline" />)).toMatchSnapshot();
    expect(renderJSON(<Box display="inline-block" />)).toMatchSnapshot();
    expect(renderJSON(<Box display="none" />)).toMatchSnapshot();
    expect(renderJSON(<Box display={['none', 'none', 'block']} theme={theme} />)).toMatchSnapshot();
  });
});
