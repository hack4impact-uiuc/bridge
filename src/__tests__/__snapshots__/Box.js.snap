// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders margin 1`] = `
.c0 {
  margin: 4px;
}

<div
  className="c0"
/>
`;

exports[`Box renders margin 2`] = `
.c0 {
  margin: 0px;
}

@media screen and (min-width:544px) {
  .c0 {
    margin: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    margin: 8px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    margin: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box renders margin 3`] = `
.c0 {
  margin: 4px;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:992px) {

}

@media screen and (min-width:544px) {
  .c0 {
    margin: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    margin: 4px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    margin: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box renders padding 1`] = `
.c0 {
  padding: 4px;
}

<div
  className="c0"
/>
`;

exports[`Box renders padding 2`] = `
.c0 {
  padding: 0px;
}

@media screen and (min-width:544px) {
  .c0 {
    padding: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding: 8px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box renders padding 3`] = `
.c0 {
  padding: 4px;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:992px) {

}

@media screen and (min-width:544px) {
  .c0 {
    padding: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding: 4px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box renders without any props 1`] = `
<div
  className=""
/>
`;

exports[`Box respects alignSelf 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

<div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-ik9o9h-0 THeBC"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Box respects display 1`] = `
.c0 {
  display: inline;
}

<div
  className="c0"
  display="inline"
/>
`;

exports[`Box respects display 2`] = `
.c0 {
  display: inline-block;
}

<div
  className="c0"
  display="inline-block"
/>
`;

exports[`Box respects display 3`] = `
.c0 {
  display: none;
}

<div
  className="c0"
  display="none"
/>
`;

exports[`Box respects display 4`] = `
.c0 {
  display: none;
}

@media screen and (min-width:544px) {
  .c0 {
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    display: block;
  }
}

<div
  className="c0"
  display={
    Array [
      "none",
      "none",
      "block",
    ]
  }
/>
`;

exports[`Box respects flex 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-flex: 4 4;
  -ms-flex: 4 4;
  flex: 4 4;
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-ik9o9h-0 etMsmz"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Box respects flexBasis 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  -webkit-flex-basis: 5rem;
  -ms-flex-preferred-size: 5rem;
  flex-basis: 5rem;
}

<div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-ik9o9h-0 kGqbKb"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Box respects flexGrow 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  -webkit-box-flex: 4;
  -webkit-flex-grow: 4;
  -ms-flex-positive: 4;
  flex-grow: 4;
}

<div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-ik9o9h-0 cvhGzj"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Box respects flexShrink 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  -webkit-flex-shrink: 4;
  -ms-flex-negative: 4;
  flex-shrink: 4;
}

<div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-ik9o9h-0 cnxAgD"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Box respects order 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  -webkit-order: 5;
  -ms-flex-order: 5;
  order: 5;
}

<div>
      <div
        class="c0"
        order="5"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-ik9o9h-0 bcDEBP"
      order="5"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
